docs/blog/2019-02-06-government-open-data-site-with-gatsby/index.md

```diff
@@ -0,0 +1,237 @@
+With our homepage rebuild, we also started using [CSS Modules](https://github.com/css-modules/css-modules). CSS Modules are CSS files "in which all class names and animation names are scoped locally by default," and they're a great way to avoid selector name collisions and limit the accumulation of unused styles.
+
+Scoping and storing each component's CSS locally makes it much easier to track down and edit a component's styling.
+
+### A modern framework
+
+Nothing is future-proof in technology, but migrating to Gatsby gets us closer, with a flexible, modular architecture and a growing and spirited community of developers behind it. By migrating to Gatsby, we're trying to build a maintainable, future-friendly codebase for the team that comes after us.
+
+## Challenges during rebuild
+
+While our experience refactoring the homepage was largely positive, we have faced some challenges migrating to Gatsby. Some of those challenges are inherent in refactoring, caused by maintaining two codebases and build processes simultaneously. Others are perhaps due to the relatively new emergence of Gatsby itself, which is actively being worked on and upgraded.
+
+Here are the main challenges we've faced.
+
+### Long build times
+
+The site has parallel builds right now, compiling both the Gatsby and remaining Jekyll parts of the site. Build times can reach 20 minutes, much of which time is consumed by building the GraphQL schema. This can slow down our design and development workflow, and occasionally results in build time-outs on the otherwise excellent Federalist.
+
+We expect we can optimize our code to mitigate our long build times, and having only one build process will help. But we are asking a lot of GraphQL, and we understand the build time is associated with the complexity of our query schema.
+
+### Redundancies
+
+Refactoring the site over time means we often have to manage content in two places. For instance, if we update content in the Jekyll part of the site, but we also have a Gatsby version in the pipeline, we need to update that, too.
+
+In some cases, we've delayed content updates or restyling, acknowledging it makes more sense to do so after the migration of that page or asset.
+
+Thankfully, we have strong communication on our team, and we work in the open, which helps us avoid more extreme problems with redundancy.
+
+### Integrating with Jekyll
+
+Related to redundancy, providing a seamless user experience between the Gatsby and Jekyll portions of the site has been a challenge. There were three issues to resolve:
+
+1. Page URL issue
+2. Gatsby prefetch error
+3. Federalist preview URL
+
+#### Page URL issue
+
+When Gatsby builds the site, it creates a `public` directory for all required site assets. We copied this directory to a `gatsby-public` folder for the Jekyll build to use. However, we didn't want our URLs to include `gatsby-public`, so we needed to add a permalink attribute to our pages' front matter to override the default path.
```

Comment:
General question - why not use the other approach (which seems simpler?) of just re-using the `public` folder for the Jekyll build?

Reply:
Guessing you want a clear way to avoid possible duplicates or one build process conflicting with the other? Where is the `Jekyll` build in `gatsby-public`?

Also - this is just a question out of curiosity, not a blocker for this PR at all!

Reply:
I'll see if @master12 wants to reply to this one. :)